production/scrapers/ohio_population.R

```diff
@@ -51,55 +51,121 @@ ohio_population_pull <- function(x, exp_date = Sys.Date()){
 }
 
 ohio_population_restruct <- function(x){
-    list(pg1 = x %>% 
-             tabulizer::extract_tables(pages = 1, area = list(c(129, 49, 753, 564))), 
-         pg2 = x %>% 
-            tabulizer::extract_tables(pages = 2, area = list(c(45, 50, 345, 565)))
-    )
+    
+    # the resturucture for this code is tricky as the file went through a
+    # pretty big change in the way that it was formatted at some point so there
+    # are two code paths the screaper could take
+    
+    # this is the set of code corresponding to the newer file structure
+    restruct_out <- list(
+        pg1c1 = x %>%
+             tabulizer::extract_tables(
+                 pages = 1,
+                 area = list(c(108.81565, 33.26855, 764.22317, 284.02295))) %>%
+            .[[1]], 
+        pg2c2 = x %>% 
+             tabulizer::extract_tables(
+                 pages = 2,
+                 area = list(c(25.87326, 31.64027, 155.97897, 287.27950))) %>%
+            .[[1]],
+        pg1c1 = x %>% 
+            tabulizer::extract_tables(
+                pages = 1,
+                area = list(c(115.7680, 313.4366, 765.1223, 566.0635))) %>%
+            .[[1]], 
+        pg2c2 = x %>% 
+            tabulizer::extract_tables(
+                pages = 2,
+                area = list(c(27.42261, 315.65104, 290.50435, 563.66939))) %>%
+            .[[1]]
+        )
+    
+    exp_cols <- c("Institution", "Count")
+    obs_cols <- colnames(restruct_out$pg1c1)
+    
+    # if these checks dont pass then revert to using the old way of extracting
+    # the data
+    if(ncol(restruct_out$pg1c1)!= 2 | !all(obs_cols == exp_cols)){
+        z <- x %>%
```

Comment:
I think everything that runs from line 89 to 121 could be its own private function, like `.pre_2022_ohio_pop_restruct` (or something less wordy). Forgive me if I'm overexplaining, but I think that could make the code more encapsulated and easier to debug -- so like as the next person comes along, they can skim over the function that does the old way of doing stuff, or alternately, if it errors, they can see a stack trace that points them to the function that does the old extraction method. And keep it as a "private" method because its more of a helper function for the class than anything public. Google's style guide says that suggests using a `.` before the method name, which is something I've tried to do if I'm making a private helper. https://google.github.io/styleguide/Rguide.html

Reply:
Yeah I like this idea!